src/core/connection-parameters.cpp

```diff
@@ -59,4 +61,38 @@ connection_parameters::connection_parameters(std::string const & fullConnectStri
 
     factory_ = &dynamic_backends::get(backendName);
     connectString_ = connectString;
+    parse();
+}
+
+void connection_parameters::parse()
+{
+    std::stringstream ssconn(connectString_);
+    while (!ssconn.eof() && ssconn.str().find('=') != std::string::npos)
+    {
+        std::string key, val;
+        std::getline(ssconn, key, '=');
+        std::getline(ssconn, val, ' ');
+
+        char delim = 0;
+        if (val.size()>0 && (val[0]=='\"' || val[0]=='\''))
+        {
+            delim = val[0];
+            std::string quotedVal = val.erase(0, 1);
+
+            if (quotedVal[quotedVal.size()-1] ==  delim)
+            {
+                quotedVal.erase(val.size()-1);
+            }
+            else // space inside value string
+            {
+                std::getline(ssconn, val, delim);
```

Comment:
This doesn't take into account escaped delimiters which, I think, are possible, e.g. `password="in \"quotes\""`.